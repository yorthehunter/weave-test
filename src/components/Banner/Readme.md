Info Banner:
```js
<Banner visible={true} type="info">Hello</Banner>
```

Warning Banner:
```js
<Banner visible={true} type="warning">Hello</Banner>
```

Error Banner:
```js
<Banner visible={true} type="error">Hello</Banner>
```

Success Banner:
```js
<Banner visible={true} type="success">Hello</Banner>
```

Banner using `content` prop:

```js
<Banner type="info" visible={true} content="Hello" />
```
