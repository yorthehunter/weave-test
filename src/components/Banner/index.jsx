import React from 'react';
import { oneOf, string, bool, node } from 'prop-types';
import { sizeModifier, alertStyleModifier } from '../utilities/util';
import Icon from '../utilities/Icon';
import CustomTag from '../utilities/CustomTag';
import '../../weave/components/banner/index.scss';

const componentPrefix = 'weave-banner';

/**
 * A banner is a notification that is usually dismissable and often appears at
 * the top of the page, just below the nav (e.g. a flash message in Rails).
 */

class Banner extends React.Component {
  constructor(props) {
    super();

    this.state = {
      weaveBannerVisible: props.visible && true,
    };

    this.removeBanner = this.removeBanner.bind(this);
  }

  removeBanner(e) {
    e.preventDefault();
    this.setState({ weaveBannerVisible: false });
  }

  iconColor(type) {
    switch (type) {
      case 'error':
      case 'success':
        return '#FFFFFF';
      case 'info':
      case 'warning':
        return '#000000';
      default:
        return '#000000';
    }
  }

  render() {
    const {
      content,
      children,
      type,
      size,
      responsive,
      dismissable,
    } = this.props;

    const componentClasses = [
      componentPrefix,
      sizeModifier(size, componentPrefix),
      alertStyleModifier(type, componentPrefix),
      responsive && `${componentPrefix}--responsive`,
    ];

    if (this.state.weaveBannerVisible) {
      return (
        <CustomTag
          {...this.props}
          componentClasses={componentClasses}
        >
          <p className={`${componentPrefix}__message`} dangerouslySetInnerHTML={{ __html: content || children }} />
          {
            dismissable &&
            <button
              style={{ cursor: 'pointer', border: 'none', backgroundColor: 'transparent' }}
              onClick={this.removeBanner}
            >
              <Icon name="close" color={this.iconColor(type)} />
            </button>
          }
        </CustomTag>
      );
    }
    return null;
  }
}

const defaultProps = {
  tag: 'div',
  type: null,
  size: null,
  content: null,
  visible: false,
  responsive: false,
  dismissable: true,
  children: null,
};

const propTypes = {
  tag: string,
  type: oneOf([
    'error',
    'success',
    'warning',
    'info',
    null,
  ]),
  size: oneOf([
    'sm',
    'md',
    'lg',
    null,
  ]),
  /** Banner content (if props.children aren't specified) */
  content: string,
  visible: bool,
  responsive: bool,
  dismissable: bool,
  /** Banner content (if props.content isn't specified) */
  children: node,
};

Banner.defaultProps = defaultProps;
Banner.propTypes = propTypes;
export default Banner;
