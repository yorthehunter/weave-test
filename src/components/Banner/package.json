{
  "name": "weave-react-banner",
  "dependencies": {
    "weave-ui-banner": "file:../../weave/components/banner"
  },
  "version": "1.0.0",
  "description": "Weave banner React component",
  "main": "index.scss",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/stitchfix/weave.git"
  },
  "keywords": [
    "css",
    "scss",
    "weave",
    "design",
    "system",
    "style",
    "guide",
    "component",
    "library",
    "ui"
  ],
  "author": "Stitch Fix",
  "license": "UNLICENSED",
  "bugs": {
    "url": "https://github.com/stitchfix/weave/issues"
  },
  "homepage": "https://github.com/stitchfix/weave#readme"
}
