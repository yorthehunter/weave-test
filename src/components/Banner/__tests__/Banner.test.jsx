import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Banner from '../';

describe('Banner renders correctly', () => {
  it('when visible with default values', () => {
    const rendered = renderer.create(<Banner visible />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('with content passed as a prop', () => {
    const rendered = renderer.create(<Banner visible content="Here's some content!" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('with content passed as a child', () => {
    const rendered = renderer.create(<Banner visible>Test content</Banner>).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

describe('Banner', () => {
  const bannerClass = '.weave-banner';
  const wrapper = mount(<Banner visible>Banner</Banner>);

  it('correctly handles banner type', () => {
    wrapper.setProps({ type: 'error' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--error')).toEqual(true);
    wrapper.setProps({ type: 'success' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--success')).toEqual(true);
    wrapper.setProps({ type: 'info' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--info')).toEqual(true);
    wrapper.setProps({ type: 'warning' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--warning')).toEqual(true);
    wrapper.setProps({ type: null });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--error')).toEqual(false);
  });

  it('removes itself when close button clicked', () => {
    expect(wrapper.find(bannerClass)).toHaveLength(1);
    wrapper.find('button').simulate('click');
    expect(wrapper.find(bannerClass)).toHaveLength(0);
  });
});

describe('Banner renders null', () => {
  it('renders null if banner is not visible', () => {
    const bannerClass = '.weave-banner';
    const wrapper = mount(<Banner>Banner</Banner>);
    expect(wrapper.find(bannerClass)).toHaveLength(0);
  });
});
