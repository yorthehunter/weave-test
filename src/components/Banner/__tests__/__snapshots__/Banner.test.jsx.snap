// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly when visible with default values 1`] = `
<div
  className="weave-banner weave-banner--info"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <p
    className="weave-banner__message"
    dangerouslySetInnerHTML={
      Object {
        "__html": null,
      }
    }
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "cursor": "pointer",
      }
    }
  >
    <svg
      height="24"
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
        fill="#000000"
      />
      <polygon
        fillOpacity="0"
        points="0 0 24 0 24 24 0 24"
      />
    </svg>
  </button>
</div>
`;

exports[`Banner renders correctly with content passed as a child 1`] = `
<div
  className="weave-banner weave-banner--info"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <p
    className="weave-banner__message"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test content",
      }
    }
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "cursor": "pointer",
      }
    }
  >
    <svg
      height="24"
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
        fill="#000000"
      />
      <polygon
        fillOpacity="0"
        points="0 0 24 0 24 24 0 24"
      />
    </svg>
  </button>
</div>
`;

exports[`Banner renders correctly with content passed as a prop 1`] = `
<div
  className="weave-banner weave-banner--info"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <p
    className="weave-banner__message"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Here's some content!",
      }
    }
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "cursor": "pointer",
      }
    }
  >
    <svg
      height="24"
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
        fill="#000000"
      />
      <polygon
        fillOpacity="0"
        points="0 0 24 0 24 24 0 24"
      />
    </svg>
  </button>
</div>
`;
